keps/sig-testing/4224-replace-kubeup-clusters-with-kops/README.md

```diff
@@ -0,0 +1,429 @@
+- We have some tests in kubernetes that make assumptions about a specific pieces of cloud infrastructure and are not a good fit for kubernetes e2e tests. Tests that rely on cloud-infra that are not reachable will be removed.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Replace kube-up clusters with kOps clusters for e2e testing
+- Ensure all e2e tests can run successfully on the kops clusters and delete/rework tests as needed
+- Delete tests that tested the bootstrap functionality of kube-up clusters. (Example: `Firewall rule should have correct firewall rules for e2e cluster`)
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Stop testing kube-dns
+- Cleanup addon-manager and related tests
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Replace kube-up clusters with kops clusters for e2e testing.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+N/A
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+N/A
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We'll create new prowjobs that use kops clusters to run cluster e2e testing. kops has been used for e2e testing for a long time but it runs a narrower set of tests designed for testing Kubernetes Distributions with various components.
+
+### Test Plan and Graduation Criteria
+
+This migration will be completed in phases:
+
+- Phase One: Ensure that functionally identical canary job of the following periodic jobs are in-place by the end of 1.29 release and are GREEN.
```

Comment:
To make it easier on sig-node, it would be nice to have a FCOS (Fedora COS) canary test.  The CRIO jobs all use this and if we don't cover that, it may make it difficult to migrate the CRIO jobs to use kops.

Reply:
node e2e tests aren't affected by this proposal, however, kops doesn't support crio today. If someone added that feature we could also add test combinations that used CRIO

Reply:
Right, node e2e is out of scope. We're trying to phase out kube-up cluster e2e.

kops fills the role kube-up cluster e2e tests have been filling. The goal is not to be the only thing we run in CI, just to replace kube-up. KIND, CAPI, etc still exist as well.

Reply:
It isn't clear reading this that node e2e tests are excluded.  In Phase 3, you will not be including node e2e tests (or sig-node) in replacing their own jobs with kops?

Reply:
Sorry, I am ignorant of these details.  It may be worth adding that in the non-goals to clarify that.